### PR TITLE
Update documentation to generate into README.

### PR DESCRIPTION
**Pull request by rwjblue**
* update docs script to inject into README
* remove old / outdated docs in README
* remove gh-pages auto-deployment

[Rendered README.md](https://github.com/rwjblue/ember-test-helpers/blob/interaction-helpers-root/README.md) for easier reviewing...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,60 +4,583 @@ A test-framework-agnostic set of helpers for testing Ember.js applications.
...
+-   `mousedown`
+-   `focus`
+-   `focusin`
+-   `mouseup`
+-   `click`
+
+The exact listing of events that are triggered may change over time as needed
+to continue to emulate how actual browsers handle clicking a given element.
+
+**Parameters**
```
Comment (Turbo87) [inline README.md]: On mobile this renders the same as the h4 of the functions which makes it somewhat hard to read and find stuff 🤔

File: README.md [hunk 2]
```diff
@@ -4,60 +4,583 @@ A test-framework-agnostic set of helpers for testing Ember.js applications.
...
 ## Collaborating
 
 ### Installation
 
-* `git clone <repository-url>` this repository
-* `cd ember-test-helpers`
-* `yarn install`
+-   `git clone <repository-url>` this repository
+-   `cd ember-test-helpers`
+-   `yarn install`
```
Comment (Turbo87) [inline README.md]: IMHO it's quite annoying that documentation is reformatting the whole file when it's only supposed to replace a single section. I've had the same issue in qunit-dom and decided to have it output in a API.md instead

File: documentation.yml [hunk 3]
```diff
@@ -1,50 +1,50 @@
...
-  - triggerKeyEvent
-  - fillIn
+    children:
+      - click
+      - tap
+      - focus
+      - blur
+      - triggerEvent
+      - triggerKeyEvent
+      - fillIn
```
Comment (Turbo87) [inline documentation.yml]: I tried this at first, but it looked strange in the HTML version. Is this still the case?

Comment (rwjblue) [reply documentation.yml]: Yes, this _only_ works for the `md` version. If you do `children:` in the HTML formatted output its _really really_ weird.

File: README.md (hunk 2 above)
Comment (rwjblue) [reply README.md]: Ya, agreed that its odd.

File: README.md (hunk 1 above)
Comment (rwjblue) [reply README.md]: Yeah, I think that is also true on desktop.

**Comment by rwjblue** [comment]
FWIW, I don't think this is a "final" solution. I just spent so much damned time on the docs, and I'd like to have _something_ to show for it while we figure out a deployment strat for the HTML formatted output...

**Comment by Turbo87** [comment]
aye, I think I'd still prefer an API.md though 🤔

**Pull request merged**